src/libraries/Delegator.sol

```diff
@@ -0,0 +1,38 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+import 'ROOT/libraries/DelegationTarget.sol';
+
+
+contract Delegator is DelegationTarget {
+    function Delegator(address _controller, bytes32 _controllerLookupName) {
+        controller = Controller(_controller);
+        controllerLookupName = _controllerLookupName;
+    }
+
+    // We currently only support a single non-array return value
+    // If we truly need to return more advanced sets of data we could follow
+    // the pattern demonstrated here: https://gist.github.com/Arachnid/4ca9da48d51e23e5cfe0f0e14dd6318f
+    // But in a way that is automated.
+    function() payable {
+        // Do nothing if we haven't properly set up the delegator to delegate calls
+        if (controllerLookupName == 0) {
+            return;
+        }
+
+        // Get the delegation target contract
+        address _target = controller.lookup(controllerLookupName);
+
+        assembly {
+            let _o_code := mload(0x40) //0x40 is the address where the next free memory slot is stored in Solidity
```

Comment:
Ah, good find on that SO article.  So if I understand this code, we are reading whatever is at 0x40, which should be a memory offset, and then writing to the offset returned by that.  Then for the return value we overwrite into the same location, because we know we don't need the space anymore and that address was a good choice before the function call so it seems reasonable that it is a good choice after the function call as well.  I wonder if we should be updating the pointer at `0x40` to be `_calldataMemoryOffset + calldatasize`?  If so, then we should probably store the return value at `_calldataMemoryOffset + calldatasize` and bump the number at 0x40 to `_calldataMemoryOffset + calldatasize + 32`.  Alternatively, we could re-use the space but "allocate" extra to ourselves by setting the value at 0x40 to `_calldataMemoryOffset + max(calldatasize, 32)`.